include/zephyr/bluetooth/conn.h

```diff
@@ -1020,15 +1020,21 @@ void bt_conn_cb_register(struct bt_conn_cb *cb);
  */
 void bt_set_bondable(bool enable);
 
-/** @brief Allow/disallow remote OOB data to be used for pairing.
+/** @brief Allow/disallow remote LE SC OOB data to be used for pairing.
  *
- *  Set/clear the OOB data flag for SMP Pairing Request/Response data.
- *  The initial value of this flag depends on BT_OOB_DATA_PRESENT Kconfig
- *  setting.
+ *  Set/clear the LE SC OOB data flag for SMP Pairing Request/Response data.
  *
- *  @param enable Value allowing/disallowing remote OOB data.
+ *  @param enable Value allowing/disallowing remote LE SC OOB data.
  */
-void bt_set_oob_data_flag(bool enable);
+void bt_set_lesc_oob_data_flag(bool enable);
+
+/** @brief Allow/disallow remote legacy OOB data to be used for pairing.
+ *
+ *  Set/clear the legacy OOB data flag for SMP Pairing Request/Response data.
+ *
+ *  @param enable Value allowing/disallowing remote legacy OOB data.
+ */
+void bt_set_legacy_oob_data_flag(bool enable);
```

Comment:
If we are already breaking the API, let's align it for consistency.
`bt_le_oob` as prefix, using the abbreviaton `sc` instead of `lesc` (see `bt_le_oob_get_sc_data` for example

Suggestion:
`bt_le_oob_set_legacy_oob_flag`,
`bt_le_oob_set_sc_oob_flag`
(maybe omit the second oob in these?)
`bt_le_oob_set_lesc